### PR TITLE
Remove unnecessary retry loop after #1220

### PR DESCRIPTION
**Pull request by ma2bd**
## Motivation

Simplify client code after #1220

## Proposal

Do not retry any more in case of an error `InactiveChain` during `handle_certificate`.

## Test Plan

CI

### CODE REVIEW

File: linera-core/src/updater.rs [hunk 1]
```diff
@@ -162,83 +162,74 @@ where
     S: Storage + Clone + Send + Sync + 'static,
     ViewError: From<S::ContextError>,
 {
-    async fn send_certificate(
+    async fn send_optimized_certificate(
```
Comment (afck) [inline linera-core/src/updater.rs:165]: So "optimized" means we don't retry if the validator is missing bytecode? Would be good to say that in a method comment.

**Review by afck — APPROVED** [review]

File: linera-core/src/updater.rs (hunk 1 above)
Comment (ma2bd) [reply linera-core/src/updater.rs:165]: I meant to say that we try the "lite" one first

Comment (ma2bd) [reply linera-core/src/updater.rs:165]: maybe "send_certificate_optimistically" is better?

Comment (afck) [reply linera-core/src/updater.rs:165]: Ah, makes sense! I'm fine with either name; I just misunderstood at first.

**Pull request merged**